x/programs/runtime/import_program.go

```diff
@@ -46,16 +70,16 @@ func NewProgramModule(r *WasmRuntime) *ImportModule {
 					context.Background(),
 					&newInfo)
 				if err != nil {
-					return nil, err
+					if code, ok := extractErrorCode(err); ok {
+						return Err[RawBytes, ErrorCode](code), nil
+					}
+					return Err[RawBytes, ErrorCode](ExecutionFailure), err
 				}
 
-				// subtract the fuel used during this call from the calling program
-				remainingFuel := newInfo.RemainingFuel()
-				if err := callInfo.ConsumeFuel(input.Fuel - remainingFuel); err != nil {
-					return nil, err
-				}
+				// return any remaining fuel to the calling program
+				callInfo.AddFuel(newInfo.RemainingFuel())
```

Comment:
I see that `RemainingFuel` is calculated from the amount of consumed fuel of the instance.
Since the reuse of instance, are we resetting the fuel metering ?

Reply:
we aren't currently reusing instances